dynawo/sources/Modeler/Common/DYNModeler.cpp

```diff
@@ -189,31 +189,52 @@ Modeler::initStaticRefs(const shared_ptr<SubModel>& model, const shared_ptr<Mode
 void
 Modeler::replaceStaticAndNodeMacroInVariableName(const shared_ptr<SubModel>& subModel1, string& var1,
     const shared_ptr<SubModel>& subModel2, string& var2) const {
-  //  @todo add same mechanism for @NODE1@, @NODE2@, @NODE3@
 
-  // convention : if node inside a connector, the staticId of the component is before @NODE@
+  // convention : if node inside a connector, the staticId of the component is placed before
+  // so the name of the var is as follow : @STATIC_ID@@@NODE@_var or @STATIC_ID@@@NODE1@_var or @STATIC_ID@@@NODE2@_var
   const string labelNode = "@NODE@";
+  const string labelNode1 = "@NODE1@";
+  const string labelNode2 = "@NODE2@";
   const string labelStaticId = "@STATIC_ID@";
 
-  // so the name of the var is as follow : @STATIC_ID@@@NODE@_var
+  // replace @STATIC_ID@ with the static id of the model where the connection should be made
   bool foundStaticIdInVar1 = (var1.find(labelStaticId) != string::npos);
   bool foundStaticIdInVar2 = (var2.find(labelStaticId) != string::npos);
-  // in connector, use static id of the model where the connection should be connected
-  // replace @STATIC_ID@ by @id@
   if (foundStaticIdInVar1)
     var1.replace(var1.find(labelStaticId), labelStaticId.size(), "@" + subModel2->staticId() + "@");
   if (foundStaticIdInVar2)
     var2.replace(var2.find(labelStaticId), labelStaticId.size(), "@" + subModel1->staticId() + "@");
 
+  // replace the static id of the model where the connection should be made with the static if of the bus for @NODE@
   bool foundNodeInVar1 = (var1.find(labelNode) != string::npos);
   bool foundNodeInVar2 = (var2.find(labelNode) != string::npos);
-
   if (foundNodeInVar1 && foundNodeInVar2) {
     throw DYNError(Error::MODELER, WrongConnectTwoUnknownNodes, subModel1->name(), var1, subModel2->name(), var2);
   } else if (foundNodeInVar1) {
-    var1 = findNodeConnectorName(var1);
+    var1 = findNodeConnectorName(var1, labelNode);
   } else if (foundNodeInVar2) {
-    var2 = findNodeConnectorName(var2);
+    var2 = findNodeConnectorName(var2, labelNode);
+  }
+
+  // replace the static id of the model where the connection should be made with the static if of the bus for @NODE1@
+  bool foundNode1InVar1 = (var1.find(labelNode1) != string::npos);
```

Comment:
Maybe we could create an utility method to avoid copy pasting 3 times almost the same algo?

Reply:
Would it be a method of the class?

Reply:
yes, a private method from this class, just to avoid copy paste (and thus ease maintenability)

Reply:
ok done!